align/schema/transistor.py

```diff
@@ -0,0 +1,52 @@
+from typing import List, Union, Dict, Optional, Tuple
+from align.schema import schema
+from pydantic import validator, ValidationError, Field
+
+
+class Transistor(schema.BaseModel):
+
+    device_type: str
+    nf: int
+
+    w: Optional[float]
+    nfin: Optional[int]
+
+    model_name: str
+
+    params: Optional[Dict[str, Union[str, int, float]]]
+
+    @validator('device_type')
+    def _validate_device_type(cls, v):
+        if v not in ['parallel', 'stack']:
+            raise ValueError(f'Device type is either parallel or stack: {v}')
+        return v
+
+    @validator('model_name')
+    def _validate_model_name(cls, v):
+        if v.startswith('n') or v.startswith('p'):
+            return v
+        else:
+            raise ValueError(f'Model name should begin with n or p: {v}')
+
+    @validator('nfin')
+    def _validate_width(cls, v, values):
+        if v is None and values['w'] is None:
+            raise ValueError('Either width or nfin should be specified')
+        return v
+
+
+class TransistorArray(schema.BaseModel):
```

Comment:
TransistorArray can be made generic to handle any kind of primitive (RES, CAP etc as well)

Would defer this to a separate PR.

Reply:
Good idea. It might be useful to have a parent schema for all elements (tx, R, L, C) to derive from.